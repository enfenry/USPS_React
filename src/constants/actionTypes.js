

//<--------------------- APPLICATIONS ------------------------>
export const READ_APPLICATIONS_SUCCESSFUL = 'READ_APPLICATIONS_SUCCESSFUL';
export const READ_APPLICATIONS_PENDING = 'READ_APPLICATIONS_PENDING';
export const READ_APPLICATIONS_FAILURE = 'READ_APPLICATIONS_FAILURE';

export const CREATE_APPLICATION_SUCCESSFUL = 'CREATE_APPLICATION_SUCCESSFUL';
export const CREATE_APPLICATION_FAILURE = 'CREATE_APPLICATION_FAILURE';

export const UPDATE_APPLICATION_SUCCESSFUL = 'UPDATE_APPLICATION_SUCCESSFUL';
export const UPDATE_APPLICATION_FAILURE = 'UPDATE_APPLICATION_FAILURE';

export const DELETE_APPLICATION_SUCCESSFUL = 'DELETE_APPLICATION_SUCCESSFUL';
export const DELETE_APPLICATION_FAILURE = 'DELETE_APPLICATION_FAILURE';

//<--------------------- CUSTOMERS ------------------------>
export const READ_CUSTOMERS_SUCCESSFUL = 'READ_CUSTOMERS_SUCCESSFUL';
export const READ_CUSTOMERS_PENDING = 'READ_CUSTOMERS_PENDING';
export const READ_CUSTOMERS_FAILURE = 'READ_CUSTOMERS_FAILURE';

export const CREATE_CUSTOMER_SUCCESSFUL = 'CREATE_CUSTOMER_SUCCESSFUL';
export const CREATE_CUSTOMER_FAILURE = 'CREATE_CUSTOMER_FAILURE';

export const UPDATE_CUSTOMER_SUCCESSFUL = 'UPDATE_CUSTOMER_SUCCESSFUL';
export const UPDATE_CUSTOMER_FAILURE = 'UPDATE_CUSTOMER_FAILURE';

export const DELETE_CUSTOMER_SUCCESSFUL = 'DELETE_CUSTOMER_SUCCESSFUL';
export const DELETE_CUSTOMER_FAILURE = 'DELETE_CUSTOMER_FAILURE';

//<--------------------- ORDERS ------------------------>
export const READ_ORDERS_SUCCESSFUL = 'READ_ORDERS_SUCCESSFUL';
export const READ_ORDERS_PENDING = 'READ_ORDERS_PENDING';
export const READ_ORDERS_FAILURE = 'READ_ORDERS_FAILURE';

//<--------------------- PRODUCTS ------------------------>
export const READ_PRODUCTS_SUCCESSFUL = 'READ_PRODUCTS_SUCCESSFUL';
export const READ_PRODUCTS_PENDING = 'READ_PRODUCTS_PENDING';
export const READ_PRODUCTS_FAILURE = 'READ_PRODUCTS_FAILURE';

//<---------------------ADDRESSES ------------------------>
export const READ_ADDRESSES_SUCCESSFUL = 'READ_ADDRESSES_SUCCESSFUL';
export const READ_ADDRESSES_PENDING = 'READ_ADDRESSES_PENDING';
export const READ_ADDRESSES_FAILURE = 'READ_ADDRESSES_FAILURE';
