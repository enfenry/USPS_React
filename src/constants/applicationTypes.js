export const ADDRESS_CHANGE = 717800000;
export const MAIL_FORWARDING = 717800001;
export const PACKAGE_SUBMISSION = 717800002;