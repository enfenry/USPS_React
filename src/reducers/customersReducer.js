import { CREATE_CUSTOMER_SUCCESSFUL, CREATE_CUSTOMER_FAILURE, READ_CUSTOMERS_SUCCESSFUL, READ_CUSTOMERS_PENDING, READ_CUSTOMERS_FAILURE, UPDATE_CUSTOMER_SUCCESSFUL, UPDATE_CUSTOMER_FAILURE, DELETE_CUSTOMER_SUCCESSFUL, DELETE_CUSTOMER_FAILURE } from '../constants/actionTypes'


export default function customersReducer(state = {}, action) {
    switch (action.type) {
        case CREATE_CUSTOMER_SUCCESSFUL: {
            let newCustomers = [...state.customers];
            newCustomers.push(action.data);
            return { ...state, customers: newCustomers, requestState: { ...state.requestState, customersCreateSuccess: true, customersCreateFailed: false, error: null } };
        }
        case CREATE_CUSTOMER_FAILURE:
            return { ...state, customers: state.customers, requestState: { ...state.requestState, customersCreateSuccess: false, customersCreateFailed: true, error: action.error } };

        case READ_CUSTOMERS_SUCCESSFUL:
            return { ...state, customers: action.data.value, requestState: { customersReadSuccess: true, customersReadPending: false, customersReadFailed: false, error: null } };
        case READ_CUSTOMERS_PENDING:
            return { ...state, requestState: { customersReadSuccess: false, customersReadPending: true, customersReadFailed: false, error: null } };
        case READ_CUSTOMERS_FAILURE:
            return { ...state, requestState: { customersReadSuccess: false, customersReadPending: false, customersReadFailed: true, error: action.error } };

        case UPDATE_CUSTOMER_SUCCESSFUL: {
            const itemIndex = state.customers.findIndex((e) => (e.contactid === action.id));
            let newCustomers = [...state.customers];
            newCustomers[itemIndex] = action.data;
            return { ...state, customers: newCustomers, requestState: { ...state.requestState, customersUpdateSuccess: true, customersUpdateFailed: false, error: null } };
        }
        case UPDATE_CUSTOMER_FAILURE:
            return { ...state, customers: state.customers, requestState: { ...state.requestState, customersUpdateSuccess: false, customersUpdateFailed: true, error: action.error } };

        case DELETE_CUSTOMER_SUCCESSFUL:
            return { ...state, customers: state.customers.filter((e) => e.contactid !== action.data), requestState: { ...state.requestState, customersDeleteSuccess: true, customersDeleteFailed: false, error: null } };
        case DELETE_CUSTOMER_FAILURE:
            return { ...state, requestState: { ...state.requestState, customersDeleteSuccess: false, customersDeleteFailed: true, error: action.error } };

        default:
            return state;
    }
}