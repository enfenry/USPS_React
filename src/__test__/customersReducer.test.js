import customersReducer from '../reducers/customersReducer';
import { READ_CUSTOMERS_SUCCESSFUL, READ_CUSTOMERS_PENDING, READ_CUSTOMERS_FAILURE } from '../constants/actionTypes';
import expect from 'expect';

describe('customers reducer', () => {

    it('should return the initial state', () => {
        expect(customersReducer(undefined, {
            type: "DUMMY_TYPE"
        })).toMatchSnapshot();
    });

    it('should handle READ_CUSTOMERS_PENDING', () => {
        expect(customersReducer({}, {
            type: READ_CUSTOMERS_PENDING
        })).toMatchSnapshot();
    });

    it('should handle READ_CUSTOMERS_FAILURE', () => {
        expect(customersReducer({}, {
            type: READ_CUSTOMERS_FAILURE
        })).toMatchSnapshot();
    });

    it('should handle READ_CUSTOMERS_SUCCESSFUL', () => {
        expect(customersReducer({}, {
            type: READ_CUSTOMERS_SUCCESSFUL,
            data: {
                value: [
                    "dummy data"
                ]
            }
        })).toMatchSnapshot();
    });
});