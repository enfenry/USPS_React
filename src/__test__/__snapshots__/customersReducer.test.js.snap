// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customers reducer should handle READ_CUSTOMERS_FAILURE 1`] = `
Object {
  "customersRequestFailed": true,
  "customersRequestPending": false,
  "customersRequestSuccess": false,
}
`;

exports[`customers reducer should handle READ_CUSTOMERS_PENDING 1`] = `
Object {
  "customersRequestFailed": false,
  "customersRequestPending": true,
  "customersRequestSuccess": false,
}
`;

exports[`customers reducer should handle READ_CUSTOMERS_SUCCESFUL 1`] = `
Object {
  "customers": Array [
    "dummy data",
  ],
  "customersRequestFailed": false,
  "customersRequestPending": false,
  "customersRequestSuccess": true,
}
`;

exports[`customers reducer should return the initial state 1`] = `Object {}`;
